src/process.rs

```diff
@@ -310,46 +198,411 @@ impl Process {
+                }
+                err => {
+                    io::Error::new(io::ErrorKind::Other, format!("{}", err))
+                }
+            }
+        })
+    }
+
+    /// Flushes the logger.
+    ///
+    /// Ignores any errors.
+    fn flush(&mut self) {
+        let _ = self.0.backend.flush();
+    }
+}
+
+
+//------------ GlobalLogger --------------------------------------------------
+
+/// The global logger.
+///
+/// A value of this type can go into a static. Until a proper logger is
+/// installed, it just writes all log output to stderr.
+struct GlobalLogger {
+    /// The real logger. Can only be set once.
+    inner: OnceCell<Logger>,
+}
+
+/// The static for the log crate.
+static GLOBAL_LOGGER: GlobalLogger = GlobalLogger::new();
+
+impl GlobalLogger {
+    /// Creates a new provisional logger.
+    const fn new() -> Self {
+        GlobalLogger { inner: OnceCell::new() }
+    }
+
+    /// Switches to the proper logger.
+    fn switch(&self, logger: Logger) {
+        if self.inner.set(logger).is_err() {
+            panic!("Tried to switch logger more than once.")
+        }
+    }
+
+    /// Performs a log rotation.
+    fn rotate(&self) -> Result<(), Failed> {
+        match self.inner.get() {
+            Some(logger) => logger.rotate(),
+            None => Ok(()),
+        }
+    }
+}
+
+
+impl log::Log for GlobalLogger {
+    fn enabled(&self, _: &log::Metadata<'_>) -> bool {
+        true
+    }
+
+    fn log(&self, record: &log::Record<'_>) {
+        match self.inner.get() {
+            Some(logger) => logger.log(record),
+            None => {
+                let _ = writeln!(
+                    io::stderr().lock(), "[{}] {}",
+                    record.level(), record.args()
+                );
+            }
+        }
+    }
+
+    fn flush(&self) {
+        if let Some(logger) = self.inner.get() {
+            logger.flush()
+        }
+    }
+}
+
+
 //------------ LogOutput -----------------------------------------------------
 
 #[derive(Debug)]
 pub struct LogOutput {
-    queue: Mutex<mpsc::Receiver<String>>,
-    current: RwLock<(Bytes, DateTime<Utc>)>,
+    queue: Arc<Mutex<String>>,
+    current: RwLock<Bytes>,
 }
 
 impl LogOutput {
-    fn new() -> (mpsc::Sender<String>, Self) {
-        let (tx, rx) = mpsc::channel();
+    fn new() -> (Arc<Mutex<String>>, Self) {
+        let queue = Arc::new(Mutex::new(String::new()));
         let res = LogOutput {
-            queue: Mutex::new(rx),
-            current: RwLock::new((
+            queue: queue.clone(),
+            current: RwLock::new(
                 "Initial validation ongoing. Please wait.".into(),
-                Utc::now()
-            ))
+            )
         };
-        (tx, res)
+        (queue, res)
     }
 
     pub fn start(&self) {
-        self.current.write().1 = Utc::now();
+        let new_string = format!(
+            "Log from validation run started at {}\n\n", Utc::now()
```

Comment:
For the file/stderr logging, local time is used, not UTC. Shouldn't we use the same for both?
(Perhaps even make this configurable? Although on the other hand, I guess any serious logging happens using syslog anyway..)